src/error.ts

```diff
@@ -124,6 +124,19 @@ export function isNetworkErrorBeforeHandshake(err: MongoNetworkError): boolean {
   return err[kBeforeHandshake] === true;
 }
 
+/**
+ * An error thrown when an operation requiring a connected client was called on a closed client.
+ * @public
+ * @category Error
+ */
+export class MongoClientClosedError extends MongoError {
```

Comment:
It's uncommon for us to define a single-purpose error like this, and I think (as I said above) we can get away with moving the logic (and creation of the error) to one location inside `getTopology`. In the future I would generally say that single purpose errors may indicate an issue in code structure